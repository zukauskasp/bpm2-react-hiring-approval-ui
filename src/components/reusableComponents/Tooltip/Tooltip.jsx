import React from "react";
import './Tooltip.scss';
import PropTypes from 'prop-types';

const Message = ({ messageType, text, position, ...otherPorps }) => {
    return <div className={`tooltip tooltip-${messageType} tooltip-${position}`} {...otherPorps}>{text} </div>
}

Message.propTypes = {
    messageType: PropTypes.oneOf(["error", "success", "default"]),
    position: PropTypes.oneOf(["bottom-left", "bottom", "bottom-right", "left", "top-left", "top", "top-right", "right"]),
    text: PropTypes.string,
}

Message.defaultProps = {
    messageType: "default",
    text: "",
    position: "top-left"
}

export default Message;