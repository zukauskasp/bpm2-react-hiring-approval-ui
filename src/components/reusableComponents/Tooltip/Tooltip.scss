.tooltip {
    position: relative;
    left: 0;
    top: 110%;
    color: white;
    min-height: 48px;
    font-size: 14px;
    width: 280px;
    padding: 0 .8rem;
    margin-top: 10px;
    padding: 15px;

    &:before {
        content: "\A";
        position: absolute;
        width: 0;
        height: 0;
    }
}

.tooltip-default {
    background: #003755;

    &:before {
        border: 10px solid #003755;
    }
}

.tooltip-error {
    background: #FBB273;

    &:before {
        border: 10px solid #FBB273;
    }
}

.tooltip-success {
    background: #09B89D;

    &:before {
        border: 10px solid #09B89D;
    }

}

.tooltip-top-left:before {
    bottom: 100%;
    border-top: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;

}

.tooltip-top:before {
    bottom: 100%;
    left: 45%;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    border-top: none;
}

.tooltip-top-right:before {
    bottom: 100%;
    left: 90%;
    border-top: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}


.tooltip-bottom-left:before {
    bottom: -20%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-bottom:before {
    bottom: -20%;
    left: 45%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-bottom-right:before {
    bottom: -20%;
    left: 90%;
    border-bottom: none;
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
}

.tooltip-right:before {
    bottom: 26%;
    left: 100%;
    border-right: none;
    border-top: 10px solid transparent;
    border-bottom: 10px solid transparent;
}

.tooltip-left:before {
    bottom: 26%;
    right: 100%;
    border-left: none;
    border-top: 10px solid transparent;
    border-bottom: 10px solid transparent;
}