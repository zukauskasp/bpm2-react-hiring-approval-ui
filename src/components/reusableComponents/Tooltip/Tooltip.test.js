import React from 'react';
import { shallow, mount } from 'enzyme';

import Tooltip from './Tooltip';

describe('Tooltip', () => {

    let wrapper;

    const props = {
        text: "This is some message"
    }


    beforeEach(() => {
        wrapper = shallow(<Tooltip {...props} />);
    })

    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        expect(shallow(<Tooltip {...props} />).debug()).toMatchSnapshot();
    });

    it('expect to render component', (() => {
        expect(wrapper.length).toEqual(1);
    }));


    it('expect to render correct props', () => {
        const defaultProps = {
            messageType: "default",
            text: "This is some message",
            position: "top-left"
        }

        expect(mount(<Tooltip {...props} />).props()).toEqual(defaultProps);
    });

    it('expect to add correct classes', () => {
        const classNames = wrapper.find('div').prop('className');

        expect(classNames.indexOf('tooltip-default') > -1).toEqual(true);
        expect(classNames.indexOf('tooltip-top-left') > -1).toEqual(true);
        expect(classNames.indexOf('tooltip-bottom-left') > -1).toEqual(false);
    });

});