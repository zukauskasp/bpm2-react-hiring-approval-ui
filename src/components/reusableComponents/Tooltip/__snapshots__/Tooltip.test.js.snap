// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip expect to match latest snapshot 1`] = `
"<div className=\\"tooltip tooltip-default tooltip-top-left\\">
  This is some message
   
</div>"
`;
